.github/workflows/package-creation-ecr.yaml

```diff
@@ -0,0 +1,61 @@
+name: package-creation-ecr
+
+on:
+  workflow_call:
+    inputs:
+      iam_role_arn:
+        description: "aws iam role arn for ecr repository access"
+        type: string
+        required: true
+      aws_account_region:
+        description: "aws account region"
+        type: string
+        required: true
+      ecr_repository:
+        description: "ecr repository name"
+        type: string
+        required: true
+      context:
+        description: "source context - path location docker build"
+        type: string
+        default: "."
+        required: false
+
+jobs:
+  publish-to-ecr:
+    name: Build and Publish to ECR
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      id-token: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: configure aws credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-skip-session-tagging: true
+          role-to-assume: ${{ inputs.iam_role_arn }}
+          role-session-name: gh-actions
+          aws-region: ${{ inputs.aws_account_region }}
+
+      - name: Login to Amazon ECR
+        id: login-ecr
+        uses: aws-actions/amazon-ecr-login@v1
+
+      - name: Extract branch name
+        uses: vazco/github-actions-branch-name@v1
+        id: branch
+
+      - name: Build and push Docker image
+        uses: docker/build-push-action@v3
+        env:
+          REGISTRY: ${{ steps.login-ecr.outputs.registry }}
+          REPOSITORY: ${{ inputs.ecr_repository }}
+          IMAGE_TAG: ${{ github.sha }}
```

Comment:
sha may be too long.

Reply:
@niroz89 Any recommendation on differentiating the image tag per build?

Reply:
What I meant is shorter sha ha ha, We can just have this, and remove the line #57
```yaml
- name: Setup image repo and tag
  run: |
    echo "IMAGE_TAG=${GITHUB_SHA::8}" >> "${GITHUB_ENV}"
  shell: bash
```

Reply:
above will add into the github env, and availabe via `${{ env.IMAGE_TAG }}`